pool.go

```diff
@@ -204,7 +205,12 @@ func (p *pool) checkPool() {
 
 // TODO: this should be replaced with a real validator once one exists from boxo.
 func (p *pool) mirrorValidator(resource string, reader io.Reader) error {
-	parse, err := path.ParsePath(resource)
+	// first get the 'path' part to remove query string if present.
+	pth, err := url.Parse(resource)
```

Comment:
@willscott What if the original request is a CAR request and NOT a block request ? Will this work ?

Reply:
yeah, it shouldn't matter- all we're doing in this step and below is extracting the first cid that appears in the path. then we read whatever we get back as a car and make sure the car has that first CID